EventLog.md

```diff
@@ -0,0 +1,397 @@
+      !(n.contains('-') || "" === n.trim) option new Sequence.Id {
+        override val oid: Observation.Id = o
+        override val name: String        = n.trim
+
+        override def toString: String =
+          s"${oid.toString}-$n"
+      }
+
+    def fromString(s: String): Option[Sequence.Id] =
+      s.lastIndexOf('-') match {
+        case -1 => None
+        case  n =>
+          val (a, b) = s.splitAt(n)
+          Observation.Id.fromString(a).flatMap { Id(_, b.drop(1)) }
+      }
+
+    def unsafeFromString(s: String): Sequence.Id =
+      fromString(s) | sys.error(s"Malformed Sequence.Id: $s")
+
+    def unapply(arg: Id): Option[(Observation.Id, String)] =
+      Some((arg.oid, arg.name))
+  }
+
+}
+````
+
+## Observation Execution Events
+
+In today's OCS, the tcc and seqexec send observation execution events to an intermediary process called the "wdba" as they slew to targets and execute sequences.  Perhaps in the future these tools will write directly to the database?  At any rate, all the events share a timestamp and an observation id but some types of events need additional information.  In the multi-sequence future, the observation id should be replaced with a sequence id for all (or maybe just most) events.
+
+These event types correspond, roughly, to what we have today:
+
+````scala
+package gem
+
+import java.time.Instant
+
+object EventLog {
+  sealed trait Event {
+    def timestamp: Instant
+    def sid: Sequence.Id
+  }
+
+  case class EventStartSlew(    timestamp: Instant, sid: Sequence.Id) extends Event
+  case class EventEndSlew(      timestamp: Instant, sid: Sequence.Id) extends Event
+  case class EventStartVisit(   timestamp: Instant, sid: Sequence.Id) extends Event
+  case class EventEndVisit(     timestamp: Instant, sid: Sequence.Id) extends Event
+  case class EventStartSequence(timestamp: Instant, sid: Sequence.Id) extends Event
+  case class EventEndSequence(  timestamp: Instant, sid: Sequence.Id) extends Event
+
+
+  case class EventPauseObserve(   timestamp: Instant, sid: Sequence.Id, why: Option[String]) extends Event
+  case class EventContinueObserve(timestamp: Instant, sid: Sequence.Id, why: Option[String]) extends Event
+  case class EventAbortObserve(   timestamp: Instant, sid: Sequence.Id, why: Option[String]) extends Event
+  case class EventStopObserve(    timestamp: Instant, sid: Sequence.Id, why: Option[String]) extends Event
+
+  case class EventStartIntegration(timestamp: Instant, sid: Sequence.Id, step: Int) extends Event
+  case class EventEndIntegration(  timestamp: Instant, sid: Sequence.Id, step: Int) extends Event
+}
+````
+
+Here I replaced "dataset" start/end events with "integration" start/end keeping in mind that multiple datasets may be produced for each step.
+
+As indicated by vertical whitespace in the code above there are three categories of events.  Correspondingly, I assume there will need to be three tables to store them (?).
```

Comment:
Not that I'm against a single table for this case because I don't understand the requirements, but wouldn't this be a case of premature optimization? It all depends of how the data will end up being accessed, if every time you access it you have to do the union query it makes sense to collapse it into one table, but if we need the flexibility to retrieve or insert data in unpredictable ways, I think it would reasonable to have separate tables. If performance is an issue we can always create an index for that query.

I lean towards doing normalization by default and only denormalize when there is a clear advantage.